clang/include/clang/CIR/Dialect/IR/CIROps.td

```diff
@@ -2821,13 +2821,16 @@ def VecSplatOp : CIR_Op<"vec.splat", [Pure,
   let description = [{
     The `cir.vec.splat` operation creates a vector value from a scalar value.
     All elements of the vector have the same value, that of the given scalar.
+    If `poison` is specified, the start value of the result is poison values
+    instead of undef.
   }];
 
-  let arguments = (ins CIR_AnyType:$value);
+  let arguments = (ins CIR_AnyType:$value, UnitAttr:$poison);
```

Comment:
I was thinking more about this, and I have an alternative suggestion: `poison` values are usually propagated in LLVM, but we don't propagate or so far intend to have similar in CIR, so adding this here would be too far off with the rest of CIR (mea culpa cause I gave you this suggestion few days ago).

So far, only these intrinsics have preferred using `poison` for these instructions, but otherwise we use undefs. I think we (a) should not add this attribute and instead (b) let this be an ABI based decision at LowerToLLVM time. For instance, when creating LLVM's insertelemtent and vectorshuffle we ask "is the type is CIR vector and is the target Neon", if so both LLVM instructions get the `poison` attribute. You should use target lowering info in LowerToLLVM to ask those questions. Let me know if you have more questions.

Reply:
Fixed, we're back to clean solution, VecSplatOP should always use PoisonOp to be consistent with OG